src/axom/quest/MeshTester.hpp

```diff
@@ -33,12 +58,284 @@ enum class WatertightStatus : signed char
+  FloatType* aabbs = axom::allocate< FloatType >( ncells * stride );
+
+  // Initialize the bounding box for each Triangle and marks
+  // if the Triangle is degenerate.
+  AXOM_PERF_MARK_SECTION( "init_tri_bb",
+    mint::for_all_cells< ExecSpace, mint::xargs::coords >(
+      surface_mesh, AXOM_LAMBDA( IndexType cellIdx,
+                                 numerics::Matrix< double >&coords,
+                                 const IndexType* AXOM_NOT_USED(nodeIds) )
+    {
+      detail::Triangle3 tri;
+
+      for ( IndexType inode=0 ; inode < 3 ; ++inode )
+      {
+        const double* node = coords.getColumn( inode );
+        tri[ inode ][ 0 ] = node[ mint::X_COORDINATE ];
+        tri[ inode ][ 1 ] = node[ mint::Y_COORDINATE ];
+        tri[ inode ][ 2 ] = node[ mint::Z_COORDINATE ];
+      } // END for all cells nodes
+
+      degenerate[ cellIdx ] = ( tri.degenerate() ? 1 : 0 );
+
+      tris[cellIdx] = tri;
+
+      detail::SpatialBoundingBox triBB = compute_bounding_box(tri);
+
+      const IndexType offset = cellIdx * stride;
+
+      xmin[cellIdx] = aabbs[ offset ]     = triBB.getMin()[0];
+      ymin[cellIdx] = aabbs[ offset + 1 ] = triBB.getMin()[1];
+      zmin[cellIdx] = aabbs[ offset + 2 ] = triBB.getMin()[2];
+
+      xmax[cellIdx] = aabbs[ offset + 3 ] = triBB.getMax()[0];
+      ymax[cellIdx] = aabbs[ offset + 4 ] = triBB.getMax()[1];
+      zmax[cellIdx] = aabbs[ offset + 5 ] = triBB.getMax()[2];
+    } );
+  );
+
+  // Copy degenerate data back to host
+  int * host_degenerate = axom::allocate< int >( ncells, 
+    getUmpireResourceAllocatorID(umpire::resource::Host) );
+  axom::copy(host_degenerate, degenerate, ncells * sizeof(int));
+
+  // Return degenerateIndices
+  for (int i = 0 ; i < ncells; i++)
+  {
+    if (host_degenerate[i] == 1)
+    {
+      degenerateIndices.push_back(i);
+    }
+  }
+
+  // Construct BVH
+  axom::spin::BVH< NDIMS, ExecSpace, FloatType > bvh( aabbs, ncells, poolID );
+  bvh.build( );
+
+  // Run find algorithm
+  IndexType* offsets    = axom::allocate< IndexType >( ncells );
+  IndexType* counts     = axom::allocate< IndexType >( ncells );
+  IndexType* candidates = nullptr;
+  bvh.findBoundingBoxes( offsets, counts, candidates, ncells, xmin, xmax,
+                         ymin, ymax, zmin, zmax );
+
+  // Get the total number of candidates
+  using REDUCE_POL =
+        typename axom::execution_space< ExecSpace >::reduce_policy;
+
+  RAJA::ReduceSum< REDUCE_POL, int> totalCandidates(0); 
+  for_all< ExecSpace >( ncells, AXOM_LAMBDA (IndexType i)
+  {
+    totalCandidates += counts[i];
+  });
+
+  //Deallocate no longer needed variables
+  axom::deallocate(aabbs);
+  
+  axom::deallocate(degenerate);
+  axom::deallocate(host_degenerate);
+
+  axom::deallocate(xmin);
+  axom::deallocate(ymin);
+  axom::deallocate(zmin);
+
+  axom::deallocate(xmax);
+  axom::deallocate(ymax);
+  axom::deallocate(zmax);
+
+  int* intersection_pairs =
+    axom::allocate <int> (totalCandidates * 2);
```

Comment:
Does this work? I thought you had to do ``totalCandidates.get()`` on the. RAJA reduction type? (but that may have changed)